lib/Core/Executor.cpp

```diff
@@ -1544,36 +1559,48 @@ void Executor::executeInstruction(ExecutionState &state, KInstruction *ki) {
 #else
           BasicBlock *caseSuccessor = si->getSuccessor(i);
 #endif
-          std::map<BasicBlock*, ref<Expr> >::iterator it =
-            targets.insert(std::make_pair(caseSuccessor,
-                           ConstantExpr::alloc(0, Expr::Bool))).first;
 
-          it->second = OrExpr::create(match, it->second);
+          // TODO add conditional
+          std::pair<std::map<BasicBlock *, ref<Expr> >::iterator, bool> res =
+              targets.insert(std::make_pair(
+                  caseSuccessor, ConstantExpr::alloc(0, Expr::Bool)));
+
+          res.first->second = OrExpr::create(match, res.first->second);
```

Comment:
@delcypher The point for this is that there could be multiple `case`s with the same basic block as destination, e.g.:

```
switch(a) {
 case '1':
 case '2':
      // do something
      break;
  default:
}
```

Therefore at this point we memorize that this value can reach the basic block including all  others found previously.